### ISSUE TITLE
[FEATURE REQUEST] Portable Apps

### ISSUE THREAD

**Issue by leminsc8er** [marticliment/UniGetUI#3102]
### Please confirm these before moving forward.

- [X] I have searched for my feature proposal and have not found a work-in-progress/duplicate/resolved/discarded issue.
- [X] This proposal is a completely new feature. If you want to suggest an improvement or an enhancement, please use [this template](https://github.com/marticliment/WingetUI/issues/new?assignees=marticliment&labels=enhancement&projects=&template=enhancement-improvement.yml&title=%5BENHANCEMENT%5D+%28Enter+your+description+here%29).

### Describe the new feature

Thank you for this great app!

i wish you will Support custom Folder and Apps.
I have some Portable Software like 7zip, WACUP, TOR, BleachBit and some more.
Also i use somtimes Portable Apps from PortableApps.com
Way i use Portable Software? 
It can be clean up your Device. eays to move software on some device. easy reinstall of backups with apdata.

if it is not posible autoupdate than just and versionscheck.
an Coummunity Database can help to add webside/orginal downloadsource and some optional target.
as soruce can be  github websidecrwaler or protableapps.com.
added manual rules by user can help to update in the future automaticly the portable version of webside, the 32 or 64bit and unpacking archivs in the right directory.

example:
- https://www.7-zip.org/a/7z2409-x64.exe (is v24.09 A rule can help to update in future https://www.7-zip.org/a/7z****-x64.exe or https://github.com/ip7z/7zip/releases (webside rule or update from gibthub)
- https://getwacup.com/ (crawl from side)
- https://github.com/thetorproject (update from github)
- https://github.com/bleachbit/bleachbit/releases (32 or 64 chose by users rule)

right now i use UChecker (Portable) to check my Portable software or seperat install folder and update manual.

i hop you understand.

### Describe how this new feature could help users

-Update check for custom folder means portable apps
- optional auto update for portable apps

**Comment by marticliment** [comment]
duplicate of https://github.com/marticliment/UniGetUI/issues/2657

**Comment by mrixner** [comment]
I actually don't think this is a duplicate of #2657; that is asking for UniGetUI to be a part of the PortableApps.com database, whereas this is asking for UniGetUI to manage portable apps.

**Comment by marticliment** [comment]
Oh, my bad.

**Comment by leminsc8er** [comment]
Yes twice.
1. i use it now portable and use scoop
But if i install scoop it will be install C:/User/
edit: if i move lokal scoop can't add lokal source in uniget for scoop.
edit: maybe https://github.com/marticliment/UniGetUI/wiki entry can also help...

2.  a modifcated PortbaleApps.com version with portable support for personal portable apps and PortableApps would be also nice.
but i want a personal install folder or same like uniget for portable workflow. it is possible for inexperienced users?
i want update PortablAapps Applications by uniceget and
some other portable apps like bleachbit Etcher and co.
Describe how this new feature could help users

thank you for this usefull opensoruce and usefull Application <3

For portable and inexperienced users experience

**Comment by AtmanActive** [comment]
+1

I was pleasantly surprised to see Portable option when executing UniGetUI installer.
That's great.
But that's just one part of the whole picture; having UniGetUI self-contained in a folder keeping all the settings close.
The next (big) part of that picture would be teaching UniGetUI how to scan for, recognize and manage portable software: software that even windows doesn't know it's there.
For this purpose, I would propose buckets: ability for UniGetUI to work with different source trees. So, the first, default bucket would be installed apps (as it is now), but then, users could add more buckets, pointing UniGetUI to a path on disk. Similar to Git repositories, these buckets wouldn't directly touch each other, and each bucket would hold data like a new UniGetUI instance was spawned.
In other words, same like how a browser can have one tab open or many tabs open, UniGetUI could have one bucket to work on (as it is now), or many buckets to work on (installed apps, portable apps on drive F, portable apps on drive G, unix-like tools in the folder xxx/yyy/zzz, and so on).

Just my train of thoughts on this subject.

I personally have 10 installed apps that WinGet/UniGetUI can see, and about a 1000 portable apps, collected from all sides of the world for the last 25 years, which, of course, current version of UniGetUI can't see.

It would be awesome if UniGetUI could break that barrier and help with managing portable software, in addition to regular, installed software.

Thanks.

**Comment by marticliment** [comment]
@AtmanActive, the package manager Scoop works in a very similar way, and has the same philosophy, perhaps you could give it a try. (https://scoop.sh)

**Comment by AtmanActive** [comment]
@marticliment : thank you for that, will definitely try Scoop.

From what little I read about it from its home page, Scoop simply installs everything in its own subfolder. Yes, that's one way to keep track of what's installed, albeit a bit too restrictive, if you ask me. Maybe there is a command line option to point it somewhere else? But then the question is, will it be able to work recursively; to traverse the given tree and find all the exes and recognize them as packages ...

I'll test that out.

Is there any way to portabilize Scoop itself under UniGetUI Portable? Same like how you added WinGet exe in a subfolder so UniGetUI can work with it right away.
If we could manually add Scoop exe to UniGetUI Portable, then, **THAT** would be an extra bucket right there: packages installed by Scoop that are installed into a subfolder of Scoop, which is by itself a subfolder of UniGetUI. Wonderful.

**Comment by marticliment** [comment]
As far as I am concerned, you can install scoop whereever you want, you just need to use the appropiate parameters on the scoop install command (please see their documentation)

**Comment by AtmanActive** [comment]
Judging by [this](https://github.com/vegardit/scoop-portable), Scoop is not portable at all.

**Comment by DantesNoWar** [comment]
Luckily, I just discovered this discussion; I was about to publish mine! Here’s my POV on why portable apps support in UniGetUI is necessary:

**My Workflow Challenges:**
I’m a heavy user of standalone portable apps and PortableApps because they offer better performance on older PCs, enable app compartmentalization for privacy, and reduce disk clutter.

**However, PortableApps has significant limitations:**

1. Limited catalog: Many apps I need to are unavailable or outdated;
2. No improvements: AFAIK The platform is barely maintained; no plans to extend the catalog or improve the client;
3. No alternatives: We're forced to manual manage excluded apps since there’s no competitors that fill the void;
4. Update delays: Official updates lag weeks or even months behind releases; sometimes it's just better to install the apps;
5. Time-consuming nature: The manual process (search, verify, download, replace files/folders) consumes significant time almost weekly.

**UGUI could address all this. My proposal:**

- Portable apps integration: UniGetUI could support official portable app repos with structured metadata or APIs, like PortableApps.com, Chocolatey, Scoop, and Winget;
- Crowdsourced portable source support: Users can submit portable app repos, both established package collections and curated individual repos, later verified by UniGetUI for QA.
- Portable-only filter: Search and select portable versions exclusively, hiding native installations and vice versa;
- Centralized management: Check and update existing apps and download portable apps in dedicated folders (e.g., C:\Portables);

**This would provide:**

1. Time savings: Eliminate weekly manual maintenance across multiple PCs;
2. Better performance: Especially on older hardware;
3. Enhanced privacy: Thanks to compartmentalization, low privileges, and easy uninstallation (just delete and forget);
4. Simplified workflow: Make portable apps as easy to maintain as installed ones;
5. Choice freedom: one can choose to ditch native installations, as maintaining non-auto-updating portable apps across multiple machines would finally be manageable and easy as maintaining installed apps.

Two simple "Ignore portable apps" and "Ignore native apps" filter toggles in the filter section would be ideal.
<img width="1920" height="1080" alt="Image" src="https://github.com/user-attachments/assets/90421d0b-a5bd-4ede-8f22-312ae36f23ab" />

> Note: This was originally prepared as a separate issue before discovering this existing discussion. I’m adding it here to give my contribution to the conversation around portable apps support. I hope this will help.
